commit 0fcba2dc609049b201e84a2998159f20e545e641 (refs/changes/67/1967/4)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2017-10-11T08:16:08+00:00 (2 years ago)
    
    MANTA-3273 "manta-adm zk fixup" broken with VMs in remote DCs

diff --git a/lib/adm.js b/lib/adm.js
index 9945601..3c2a117 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -3959,7 +3959,8 @@ maAdm.prototype.auditZkServers = function ()
 
 		serverUuid = zkinstance.params['server_uuid'];
 		if (!self.ma_vms.hasOwnProperty(uuid)) {
-			if (self.ma_cns[serverUuid] !== null) {
+			if (self.ma_cns.hasOwnProperty(serverUuid) &&
+			    self.ma_cns[serverUuid] !== null) {
 				validationErrors.push(new VError(
 				    'nameservice instance "%s": VM appears ' +
 				    'to have been provisioned in this ' +
diff --git a/test/tst.zk.js b/test/tst.zk.js
index bbe3d85..ccc8b84 100644
--- a/test/tst.zk.js
+++ b/test/tst.zk.js
@@ -112,8 +112,6 @@ function runTestCase(testcase, callback)
 				    'server_uuid': server_uuid
 				};
 			}
-		} else {
-			deployed.cns[server_uuid] = null;
 		}
 
 		deployed.instances[svcid].push({
